src/main/java/com/ning/http/client/providers/grizzly/AsyncHttpClientFilter.java

```diff
@@ -474,22 +475,28 @@ private void addAcceptHeaders(final HttpRequestPacket requestPacket) {
 
     private void addCookies(final Request request, final HttpRequestPacket requestPacket) {
         final Collection<Cookie> cookies = request.getCookies();
-        if (MiscUtils.isNonEmpty(cookies)) {
+        List<org.glassfish.grizzly.http.Cookie> finalCookiesList = new ArrayList<>();
+        for (final Cookie cookie : cookies) {
+            if (!cookie.hasExpired()) {
+                finalCookiesList.add(new org.glassfish.grizzly.http.Cookie(cookie.getName(), cookie.getValue()));
+            }
+        }
+        if (MiscUtils.isNonEmpty(finalCookiesList)) {
             StringBuilder sb = new StringBuilder(128);
-            org.glassfish.grizzly.http.Cookie[] gCookies = new org.glassfish.grizzly.http.Cookie[cookies.size()];
-            convertCookies(cookies, gCookies);
+            org.glassfish.grizzly.http.Cookie[] gCookies = new org.glassfish.grizzly.http.Cookie[finalCookiesList.size()];
+            convertCookies(finalCookiesList, gCookies);
             CookieSerializerUtils.serializeClientCookies(sb, gCookies);
             requestPacket.addHeader(Header.Cookie, sb.toString());
         }
     }
 
-    private void convertCookies(final Collection<Cookie> cookies, final org.glassfish.grizzly.http.Cookie[] gCookies) {
+    private void convertCookies(final List<org.glassfish.grizzly.http.Cookie> cookies, final org.glassfish.grizzly.http.Cookie[] gCookies) {
```

Comment:
This method was used to convert from com.ning.http.client.cookie.Cookie to org.glassfish.grizzly.http.Cookie.
It's no longer needed since you are doing this conversion in the loop that filters the expired cookies